rootfs/api/models/app.py

```diff
@@ -457,14 +496,23 @@ def deploy(self, release):
                 'build_type': release.build.type,
                 'healthcheck': release.config.healthcheck,
                 'routable': routable,
-                'deploy_batches': batches
+                'deploy_batches': batches,
+                'deploy_timeout': 120,  # 2 minutes
+                'deployment_history_limit': deployment_history,
+                'deployments': deployments,
+                'release_summary': release.summary
             }
 
         # Sort deploys so routable comes first
         deploys = OrderedDict(sorted(deploys.items(), key=lambda d: d[1].get('routable')))
 
         for scale_type, kwargs in deploys.items():
             try:
+                # Is there an existing deployment in progress?
+                name = self._get_job_id(scale_type)
+                if not force_deploy and release.deployment_in_progress(self.id, name):
```

Comment:
seems like the `deployment_in_progress` function uses an in-memory boolean, which would mean that we can't run multiple controller replicas. I know we have other functionality that uses in-memory state, so this particular one isn't a dealbreaker, but do you have any aspirations or plans to refactor to be able to run 2+ replicas in the future?